modules/new_profile/php/new_profile.class.inc

```diff
@@ -114,28 +114,13 @@ class New_Profile extends \NDB_Form
         if ($request->getMethod() === "POST") {
             $this->save();
             if (!empty($this->_error)) {
-                return (new \LORIS\Http\Response())
-                    ->withHeader("Content-Type", "application/json")
-                    ->withStatus(400)
-                    ->withBody(
-                        new \LORIS\Http\StringStream(
-                            json_encode($this->_error)
-                        )
-                    );
+                return new \LORIS\Http\Response($this->_error);
```

Comment:
This doesn't change the type, but loses a lot of information (such as the HTTP response code and content type..)

Reply:
You're right, this should be BadRequest.
```suggestion
                return new \LORIS\Http\Response\JSON\BadRequest($this->_error);
```